common/src/main/scala/com/gu/multimedia/storagetier/vidispine/VidispineCommunicator.scala

```diff
@@ -239,6 +236,46 @@ class VidispineCommunicator(config:VidispineConfig) (implicit ec:ExecutionContex
     ))
   }
 
+  def getFilesOfProject(projectId: Int, pageSize: Int = 100): Future[Seq[VSOnlineOutputMessage]] = {
+      recursivelyGetFilesOfProject(projectId, pageSize = pageSize).map(_.collect({case Some(t) => t}))
+  }
+
+  def recursivelyGetFilesOfProject(projectId: Int, start: Int = 1, pageSize: Int = 100, existingResults: Seq[Option[VSOnlineOutputMessage]] = Seq()): Future[Seq[Option[VSOnlineOutputMessage]]] = {
+    getPageOfFilesOfProject(projectId, start, pageSize).flatMap(results => {
+      if (results.isEmpty || start > 10000) {
+        Future(existingResults ++ results)
```

Comment:
Would be worth outputting a `logger.warning` if we bail because `start > 10000`

Would also be worth setting the `10000` value as a constant at the top, just to make it easier to find

Reply:
Yep x 2